Source/Engine/Syntax/Syntax.cs

```diff
@@ -18,6 +19,11 @@ public override string ToString()
             return result;
         }
 
+        public Slice TextSlice { get; set; }
+        public int StartLine { get; set; }
+        public int StartCharacter { get; set; } 
```

Comment:
The syntax tree consumes much memory, so it makes sense to remove all duplicate and redundant information from syntax nodes. For example, it's sufficient to store just start and end positions in text inside Syntax nodes. There should be no references to source text string, which is the same for all nodes. There is no need for line numbers and character positions in lines, since this information could be obtained from start and end positions. 

The translation from start and end positions to line number and character position in line should be done outside of syntax parser via special line number map, which is the ordered array of line start positions in source text. In this case translation means binary search of syntax node start position in the ordered array to obtain line position.